src/common/SurgePatch.cpp

```diff
@@ -531,15 +531,37 @@ SurgePatch::SurgePatch(SurgeStorage *storage)
         }
     } lfoPhaseName;
 
+    // Assign the dynamic deactivation handlers
+    static struct OscAudioInDeact : public ParameterDynamicDeactivationFunction
+    {
+        // TODO: Don't forget to add osc phase here once we add it in XT 2.0!
+        const bool getValue(const Parameter *p) const override
+        {
+            auto cge = p->ctrlgroup_entry;
+            auto osc = &(p->storage->getPatch().scene[p->scene - 1].osc[cge]);
+            auto res = osc->type.val.i == ot_audioinput;
+
+            osc[cge].pitch.deactivated = res;
```

Comment:
Do you need to do these modifications here? I would have thought the return value was enough if you consistently use 'appears_deactivated' vs '.deactivated'?

Reply:
Good point. Will try!

Reply:
Indeed that does work. Thanks!